frontend/packages/shared/src/components/LanguageSelect/LanguageSelect.tsx

```diff
@@ -22,57 +21,40 @@ export const languageToComboBoxOption = (
 });
 
 export const LanguageSelect = ({
-  excludedLanguage,
   languages,
+  excludedLanguage,
+  primaryLanguages,
   translateLanguage,
   ...rest
 }: LanguageSelectProps &
   Omit<ComboBoxProps, 'values'> &
   AutoCompleteComboBox) => {
-  // Helpers
-  const filterSelectedLang = (
-    excludedLanguage: string | undefined,
-    valuesArray: Array<ComboBoxOption>,
-    primaryLanguages: string[]
-  ): Array<ComboBoxOption> => {
-    const valuesArrayWithoutSelectedLang = valuesArray.filter(
-      ({ value }) => value !== excludedLanguage
-    );
-    if (excludedLanguage && !primaryLanguages.includes(excludedLanguage)) {
-      return valuesArrayWithoutSelectedLang.filter(({ value }) =>
-        primaryLanguages.includes(value)
-      );
-    }
-
-    return valuesArrayWithoutSelectedLang;
-  };
-
-  const values = languages.map((l) =>
-    languageToComboBoxOption(translateLanguage, l)
-  );
-
-  const filteredValuesArray = filterSelectedLang(
-    excludedLanguage,
-    values,
+  const primaryOptions: Array<ComboBoxOption> =
     primaryLanguages
-  );
-  const optionValuesToShow = sortOptionsByLabels(filteredValuesArray);
-  // Sort option value pairs into order set in primaryOptions parameter
-  const primaryValues = optionValuesToShow
-    .filter(({ value }) => {
-      return primaryLanguages.indexOf(value) >= 0;
-    })
-    .sort((a, b) => {
-      return (
-        primaryLanguages.indexOf(a.value) - primaryLanguages.indexOf(b.value)
-      );
-    });
-
-  // Merge sorted primaryOptions and sorted values
-  const valuesToShow = [
-    ...primaryValues,
-    ...optionValuesToShow.filter((value) => !primaryValues.includes(value)),
+      ?.filter((language) => language !== excludedLanguage)
+      .map((language) =>
+        languageToComboBoxOption(translateLanguage, language)
+      ) || [];
+
+  let secondaryOptions: Array<ComboBoxOption> = [];
+
+  if (
+    !excludedLanguage ||
+    !primaryLanguages ||
+    primaryLanguages.includes(excludedLanguage)
+  ) {
```

Comment:
Tämä koko if-ehto musta vähän huono ja mahdollisesti väärä paikka tälle toteutukselle. Tämän tarkoituksena on siis se, että mikäli käyttäjä valitsee toiseksi kielivalinnaksi ensin jonkun ei-kotimaisen kielen, niin toisena kielivalintana pitäisi olla vain kotimaiset kielet tarjolla. Ja siinä tapauksessa `secondaryOptions` jää täällä tyhjäksi arrayksi.

Jos tämän if-blokin poistaisi täältä ja `secondaryOptions` asetettaisiin aina kuten tämän sisällä tehdään, jäisi AKR:n vastuulle pitää huolta siitä, että muita kielivalintoja ei tuossa tapauksessa näytettäisi. `languages` propsissa pitäisi AKR:n komponenttien siis passsata sama `primaryLanguages` array jos näin. Musta kuulostaisi fiksummalta toteutustavalta, koska silloin `primaryLanguages` arrayn funktio olisi nimensä mukaisesti vain palauttaa ensin siinä valitut kielet (ja siinä määritellyn järjestyksen mukaan) ja sitä voisi tarvittaessa OTR:n päässäkin hyödyntää.

Yksi vaihtoehto olisi myös, että tälle `LanguageSelect` komponentille voisi antaa propseissa myös jonkun lisämuuttujan tyyliin `requirePrimaryLanguageSelection` tms. mutta ei tunnu hyvältä sekään.

Reply:
Siitä oon jokseenkin samoilla linjoilla että tää komponentti yrittää olla turhan nokkela. En saanut ihan kiinni siitä mitä tarkalleen ehdotat - miten esim. tuo `interface LanguageSelectProps` muuttuisi ehdotuksessasi?

Reply:
Olisiko siis seuraava jotakuinkin mitä ajattelit?
```
interface LanguageSelectProps {
  // If defined, excludedLanguage will not be selectable
  excludedLanguage?: string;
  // primaryLanguages (with excludedLanguage removed, if defined) will be the first selectable
  // options, in their given order
  primaryLanguages: Array<string>;
  // All selectable languages.
  // The selectable languages will be rendered as follows:
  // 1. Primary languages, in their given order, with excludedLanguage removed if defined.
  // 2. Languages, with primary languages and excludedLanguage removed, sorted by their display value.
  languages: Array<string>;
  // Other props as necessary
}
```

Reply:
Vai kenties jotain muuta? Jotain vielä enemmän yksinkertaistettuja malleja kanssa pohdin, mutta sitten ajattelin että turha tästä komponentista on kaikkea logiikkaa poistaa.

Reply:
Tuolla 

> Olisiko siis seuraava jotakuinkin mitä ajattelit?
> 
> ```
> interface LanguageSelectProps {
>   // If defined, excludedLanguage will not be selectable
>   excludedLanguage?: string;
>   // primaryLanguages (with excludedLanguage removed, if defined) will be the first selectable
>   // options, in their given order
>   primaryLanguages: Array<string>;
>   // All selectable languages.
>   // The selectable languages will be rendered as follows:
>   // 1. Primary languages, in their given order, with excludedLanguage removed if defined.
>   // 2. Languages, with primary languages and excludedLanguage removed, sorted by their display value.
>   languages: Array<string>;
>   // Other props as necessary
> }
> ```

Joo, toi taitaa olla just niin kuin meinasin että tää vois toimia. Nämä kommentit vois olla hyvä olla koodissa myös mukana.

Reply:
Toi vois olla toimiva. Varmaan kohtuu selkeä sekä shared-kirjaston että kutsujien puolesta.